tests/MongoDB.Driver.Tests/JsonDrivenTests/JsonDrivenCommandTest.cs

```diff
@@ -50,29 +51,69 @@ protected override void AssertException()
                     }
                 }
 
-                if (actualErrorCodeName == expectedErrorCodeName)
+                if (actualErrorCodeName == null)
                 {
-                    return;
```

Comment:
The flow of this method has been changed from early returns to flowing through to the end because we now assert against multiple aspects of the expected exception.